python/tvm/script/parser/core/entry.py

```diff
@@ -77,4 +95,15 @@ def parse(program: Union[doc.AST, Any, str], extra_vars: Dict[str, Any] = None)
             parser.parse(extra_vars=extra_vars)
         except ParserError as err:
             parser.report_error(err.node, err.args[0])
-    return builder.get()
+    ret = builder.get()
+    # well-formedness check will ignore any non-Relax functions
+    if isinstance(ret, IRModule):
+        # note: use the walrus operator (:=) once the project Python version
+        # supports it, would be more concise
+        source_ast = source.as_ast()
+        if find_decorator_annotation(source_ast, "check_well_formed") and not well_formed(ret):
```

Comment:
Can we add a unit test validated that `@I.ir_module(check_well_formed=False)` disables the well-formed check?  This is the only location where I see `check_well_formed` in this PR.

Reply:
Oh I thought I had gotten rid of it. It doesn't actually work because the parser, for whatever reason, does not save a decorator list for the IRModule and I don't know how to change it so that it does. I would prefer having it, but the present parser can't save that information.

Reply:
Mmm, handling the failing test cases will be a lot harder without this capability so I will see how I can improve the parser.

Reply:
Ah, I think I know why: Unlike the decorators for TIR and Relax functions (which are inside the module), the decorator for the module itself will not be saved when the module is parsed (i.e., the raw AST sent to the parser will not include the decorator because the decorator will have been "applied") so I'll have to find another means of passing that information along.